src/gui/mainwindow.h

```diff
@@ -258,6 +260,7 @@ private slots:
 #endif
     bool m_hasPython;
     QMenu *m_toolbarMenu;
+    bool m_sessionStarted;
```

Comment:
I would add `Session::isStarted() const { return m_started; }` getter and use it where needed instead of having separate variables.